src/relayedTransactionV1Builder.ts

```diff
@@ -0,0 +1,113 @@
+import {Transaction} from "./transaction";
+import {IAddress, INonce} from "./interface";
+import {INetworkConfig} from "./interfaceOfNetwork";
+import {ErrInvalidRelayedV1BuilderArguments} from "./errors";
+import {TransactionPayload} from "./transactionPayload";
+import {ContractFunction, StringValue} from "./smartcontracts";
+import {Address} from "./address";
+import BigNumber from "bignumber.js";
+
+export class RelayedTransactionV1Builder {
+    innerTransaction: Transaction | undefined;
+    relayerAddress: IAddress | undefined;
+    relayerNonce: INonce | undefined;
+    netConfig: INetworkConfig | undefined;
+
+    /**
+     * Sets the inner transaction to be used. It has to be already signed.
+     *
+     * @param {Transaction} transaction The inner transaction to be used
+     */
+    setInnerTransaction(transaction: Transaction): RelayedTransactionV1Builder {
+        this.innerTransaction = transaction;
+        return this;
+    }
+
+    /**
+     * Sets the network config to be used for building the relayed v1 transaction
+     *
+     * @param {INetworkConfig} netConfig The network configuration to be used
+     */
+    setNetworkConfig(netConfig: INetworkConfig): RelayedTransactionV1Builder {
+        this.netConfig = netConfig;
+        return this;
+    }
+
+    /**
+     * Sets the address of the relayer (the one that will actually pay the fee)
+     *
+     * @param relayerAddress
+     */
+    setRelayerAddress(relayerAddress: IAddress): RelayedTransactionV1Builder {
+        this.relayerAddress = relayerAddress;
+        return this;
+    }
+
+    /**
+     * (optional) Sets the nonce of the relayer
+     *
+     * @param relayerNonce
+     */
+    setRelayerNonce(relayerNonce: INonce) : RelayedTransactionV1Builder {
+        this.relayerNonce = relayerNonce;
+        return this;
+    }
+
+    /**
+     * Tries to build the relayed v1 transaction based on the previously set fields
+     *
+     * @throws ErrInvalidRelayedV1BuilderArguments
+     * @return Transaction
+     */
+    build(): Transaction {
```

Comment:
Here it's perfectly fine that we return the concrete `Transaction`, not an interface :+1: